timelock-agent/src/main/java/com/palantir/timelock/paxos/PaxosRemotingUtils.java

```diff
@@ -65,8 +66,8 @@ public static List<String> addProtocols(ClusterConfiguration cluster, List<Strin
 
     public static URL convertAddressToUrl(ClusterConfiguration cluster, String address) {
         try {
-            return UriBuilder.fromPath(addProtocol(cluster, address)).build().toURL();
-        } catch (MalformedURLException e) {
+            return new URIBuilder(addProtocol(cluster, address)).build().toURL();
```

Comment:
Aren't there some built in parsers?

Reply:
What do you mean?

Reply:
```UriBuilder.fromPath(addProtocol(cluster, address)).build().toURL()``` doesn't actually REALLY use the UriBuilder, you're doing manual string appending, then PARSE into URI and then convert to URL. Isn't there a built in way to go from string to URL?